pulp_docker/app/serializers.py

```diff
@@ -54,3 +60,125 @@ class Meta:
     class Meta:
         fields = platform.PublisherSerializer.Meta.fields
         model = models.DockerPublisher
+
+
+class DockerDistributionSerializer(platform.ModelSerializer):
+    _href = platform.IdentityField(
+        view_name='docker-distributions-detail'
+    )
+    name = serializers.CharField(
+        help_text=_('A unique distribution name. Ex, `rawhide` and `stable`.'),
+        validators=[validators.MaxLengthValidator(
+            models.DockerDistribution._meta.get_field('name').max_length,
+            message=_('Distribution name length must be less than {} characters').format(
+                models.DockerDistribution._meta.get_field('name').max_length
+            )),
+            UniqueValidator(queryset=models.DockerDistribution.objects.all())]
+    )
+    base_path = serializers.CharField(
+        help_text=_('The base (relative) path component of the published url. Avoid paths that \
+                    overlap with other distribution base paths (e.g. "foo" and "foo/bar")'),
+        validators=[validators.MaxLengthValidator(
+            models.DockerDistribution._meta.get_field('base_path').max_length,
+            message=_('Distribution base_path length must be less than {} characters').format(
+                models.DockerDistribution._meta.get_field('base_path').max_length
+            )),
+            UniqueValidator(queryset=models.DockerDistribution.objects.all()),
+        ]
+    )
```

Comment:
I think the returned base_url is wrong:

~/d/devel ❯ http --auth admin:admin http://dev.pulp3:8000/pulp/api/v3/docker-distributions/ base_path='foo' publication='/pulp/api/v3/publications/1/' name='foo-barname'
HTTP/1.1 201 Created
Allow: GET, POST, HEAD, OPTIONS
Content-Length: 273
Content-Type: application/json
Date: Tue, 20 Nov 2018 21:37:10 GMT
Location: /pulp/api/v3/docker-distributions/1/
Server: WSGIServer/0.2 CPython/3.6.5
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "_href": "/pulp/api/v3/docker-distributions/1/",
    "base_path": "foo",
    "base_url": "dev.pulp3:8000/pulp/content/foo",
    "content_guard": null,
    "created": "2018-11-20T21:37:10.969749Z",
    "name": "foo-barname",
    "publication": "/pulp/api/v3/publications/1/",
    "publisher": null,
    "repository": null

Reply:
Yup, base_url should be dev.pulp3:8000/foo

Reply:
Thinking a little more about this, I think we ought to not return a `base_url` at all. Instead we should have another field `docker_client_url`. If you would prefer to write an issue, I don't think it's necessary to do that in this PR.

Reply:
The docker pull command takes the following arguments:

`docker pull [-a|--all-tags] [--help] NAME[:TAG] | [REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG]`

what if we call this field 'registry_url' ?

Reply:
I think `registry_path` would be better, since thats what they seem to call it https://docs.docker.com/engine/reference/commandline/pull/#pull-from-a-different-registry